models/path.py

```diff
@@ -48,11 +48,13 @@ def is_valid(self, switch_a, switch_z, is_scheduled=False):
         """Check if this is a valid path."""
         if not self:
             return True
-        previous = switch_a
+        previous = {switch_a}
```

Comment:
Although initially, [I also had the impression that we could pull off accepting any order of the endpoints](https://github.com/kytos-ng/kytos/pull/329#issuecomment-1425958279) when validation a previous it can result in a non contiguous path, for instance a ring topology:

```
{
    "name": "epl_static",
    "dynamic_backup_path": false,
    "uni_a": {
        "interface_id": "00:00:00:00:00:00:00:01:1"
    },
    "uni_z": {
        "interface_id": "00:00:00:00:00:00:00:03:1"
    },
    "primary_path": [
        {
            "endpoint_a": {
                "id": "00:00:00:00:00:00:00:01:3"
            },
            "endpoint_b": {
                "id": "00:00:00:00:00:00:00:02:2"
            }
        },
        {
            "endpoint_a": {
                "id": "00:00:00:00:00:00:00:01:4"
            },
            "endpoint_b": {
                "id": "00:00:00:00:00:00:00:03:3"
            }
        }
    ]
}
```

Which would result in a broken dataplane since it would forward traffic to s2, which would end up being dropped, and exceptionally in this case the last link was also valid connecting to the last switch, so the path isn't actually contiguous:

![20230621_133104](https://github.com/kytos-ng/mef_eline/assets/1010796/5ba75323-6c70-4ed7-a288-32f40434b2ae)

So, the order of the endpoints actually matter to validate if the path is contigious, good thing though that the links will have deterministic endpoint with your contribution, so it's a just a matter of adjusting the algorithm here again, could you adapt it? Let's also cover this case with a unit test.

Reply:
Also, since having deterministic `Link` endpoints will lose the semantics of ordering, maybe it's not worth pursuing this new approach and maintain as it is and not fix https://github.com/kytos-ng/kytos/issues/221 in the first place? As @italovalcy mentioned in the issue the impact was just mostly convenience. Once links [are loaded they'll maintain the order they we're first created](https://github.com/kytos-ng/topology/blob/master/main.py#L125). So, still having this instantiation order of links still allow mef eline to perform these static validations.

Maybe it's worth to maintain as it is, let me know what you guys think? Unless @Alopalao you can spot another solution without having to call pathfinder for static paths.

Reply:
Fixed false positive for paths. I think the proposed changed can work as long as the path is analyzed from start/end point in order to determine whether a path is connected.

Reply:
Good one @Alopalao let's continue the thread on this commment here https://github.com/kytos-ng/mef_eline/pull/339#discussion_r1237395453